### ISSUE TITLE
Problems of Investment of MolCLR Representation part

### ISSUE THREAD

**Issue by zqwz909** [yuyangw/MolCLR#21]
Hello, I tried to do the experiment in the Investment of MolCLR Representation part of your paper, but the code did not provide the vector to generate the molecular representation. In the encoder part, I used the representation vector generated by the pre-trained gcn, and the result of cosine similarity was very incorrect. How do you use MolCLR to generate molecular representations in this part of the experiment?

**Comment by yuyangw** [comment]
Hi, thanks for your interest in our work. 
If I understand your question correctly, you want to extract the representation from GNN. The representations for input molecules are returned as h [(https://github.com/yuyangw/MolCLR/blob/master/models/gcn_molclr.py#L158)](https://github.com/yuyangw/MolCLR/blob/master/models/gcn_molclr.py#L158). 
Hope this helps. 

Best, 
Yuyang

**Comment by zqwz909** [comment]
Thank you for pointing out. I have another question. Is the feature vector obtained by the gnn network directly used to compare the cosine distance, or does it need to be normalized?

**Comment by yuyangw** [comment]
It should be normalized.

**Comment by zqwz909** [comment]
OK, can you tell me the normalization method you adopted? This part of the experiment in the paper doesn't seem to mention.Thank you!

**Comment by yuyangw** [comment]
You can refer to the cosine similarity function in pytorch: [https://pytorch.org/docs/stable/generated/torch.nn.functional.cosine_similarity.html](https://pytorch.org/docs/stable/generated/torch.nn.functional.cosine_similarity.html), which already implements normalization.

**Comment by RichardLrx** [comment]
Hello, I used the pre-training model ([pretrained_gin/checkpoints](https://github.com/yuyangw/MolCLR/tree/master/ckpt/pretrained_gin/checkpoints)) provided by you and achieved good results in my work. Therefore, I would like to know which data augmentations technology you used to get the pre-training model pretrained_gin. Is atom masking? bond deletion? subgraph removal?

**Comment by yuyangw** [comment]
The checkpoint is trained with subgraph removal since it achieves the best overall performance as reported in our paper.

**Comment by RichardLrx** [comment]
Thank you!
